src/java.base/unix/native/criuengine/criuengine.cpp

```diff
@@ -63,6 +66,52 @@ static int kickjvm(pid_t jvm, int code) {
     return 0;
 }
 
+static std::string join_args(const std::vector<const char *> &args) {
+    std::string retval;
+    for (const char *s : args) {
+        if (!s) {
+            continue;
+        }
+        if (!retval.empty()) {
+            retval += ' ';
+        }
+        // https://unix.stackexchange.com/a/357932/296319
+        if (!strpbrk(s, " \t\n!\"#$&'()*,;<=>?[\\]^`{|}~")) {
+            retval += s;
+            continue;
+        }
+        retval += '\'';
+        for (; *s; ++s) {
+            if (*s != '\'') {
+                retval += *s;
+            } else {
+                retval += "'\\''";
+            }
+        }
+        retval += '\'';
```

Comment:
It will be surprised to meet these symbols in the args, and IMHO this is overly complicated way to report them

Reply:
Should it therefore abort when it sees such symbol? Otherwise the problem will not be reproducible from the error message.

Reply:
You mean the line is not copy-pastable. But it is not such anyway, as it makes sense only for specific pid, and in the specific point in time (in VM_Crac safepoint).

I'm fine with the current code.